lib/bibdata_rs/src/theses/dataspace/document/builder.rs

```diff
@@ -0,0 +1,208 @@
+        if let Some(ref mut vec) = self.identifier_other {
+            vec.push(identifier_other.into())
+        } else {
+            self.identifier_other = Some(vec![identifier_other.into()]);
+        };
+        self
+    }
+
+    pub fn with_identifier_uri(mut self, identifier_uri: impl Into<String>) -> Self {
+        if let Some(ref mut vec) = self.identifier_uri {
+            vec.push(identifier_uri.into())
+        } else {
+            self.identifier_uri = Some(vec![identifier_uri.into()]);
+        };
+        self
+    }
+
+    pub fn with_language_iso(mut self, language_iso: impl Into<String>) -> Self {
+        if let Some(ref mut vec) = self.language_iso {
+            vec.push(language_iso.into())
+        } else {
+            self.language_iso = Some(vec![language_iso.into()]);
+        };
+        self
+    }
+
+    pub fn with_location(mut self, location: impl Into<String>) -> Self {
+        if let Some(ref mut vec) = self.location {
+            vec.push(location.into())
+        } else {
+            self.location = Some(vec![location.into()]);
+        };
+        self
+    }
+
+    pub fn with_mudd_walkin(mut self, mudd_walkin: impl Into<String>) -> Self {
+        if let Some(ref mut vec) = self.mudd_walkin {
+            vec.push(mudd_walkin.into())
+        } else {
+            self.mudd_walkin = Some(vec![mudd_walkin.into()]);
+        };
+        self
+    }
+
+    pub fn with_rights_access_rights(mut self, rights_access_rights: impl Into<String>) -> Self {
+        if let Some(ref mut vec) = self.rights_access_rights {
+            vec.push(rights_access_rights.into())
+        } else {
+            self.rights_access_rights = Some(vec![rights_access_rights.into()]);
+        };
+        self
+    }
+
+    pub fn with_title(mut self, title: impl Into<String>) -> Self {
+        if let Some(ref mut vec) = self.title {
+            vec.push(title.into())
+        } else {
+            self.title = Some(vec![title.into()]);
+        };
+        self
+    }
+
+    pub fn build(self) -> DataspaceDocument {
+        DataspaceDocument {
+            id: self.id,
+            certificate: self.certificate,
+            contributor: self.contributor,
+            contributor_advisor: self.contributor_advisor,
+            contributor_author: self.contributor_author,
+            date_classyear: self.date_classyear,
+            description_abstract: self.description_abstract,
+            department: self.department,
+            embargo_lift: self.embargo_lift,
+            embargo_terms: self.embargo_terms,
+            format_extent: self.format_extent,
+            identifier_other: self.identifier_other,
+            identifier_uri: self.identifier_uri,
+            language_iso: self.language_iso,
+            location: self.location,
+            mudd_walkin: self.mudd_walkin,
+            rights_access_rights: self.rights_access_rights,
+            title: self.title,
+        }
+    }
+}
```

Comment:
Add a test just so future tests are easier to write for future us? (code is exercised in tests in other modules)